tutorials/developers/dev_nn_datapoint/dev_nn_datapoint.tex

```diff
@@ -0,0 +1,2454 @@
+		[BRAPH2.STR ':NNDataPoint_CON_CLA:' BRAPH2.FAIL_TEST], ...
+		'NNDataPoint_CON_CLA does not construct the dataset correctly. The number of the inputs should be the same as the number of imported subjects of group 2.' ...
+		)
+
+% Check whether the number of targets matches ¥\circled{9}\circlednote{9}{tests the number of targets from the dataset matches the number of subjects in the group.}¥
+assert(length(d1.get('TARGETS')) == gr1.get('SUB_DICT').get('LENGTH'), ...
+		[BRAPH2.STR ':NNDataPoint_CON_CLA:' BRAPH2.FAIL_TEST], ...
+		'NNDataPoint_CON_CLA does not construct the dataset correctly. The number of the targets should be the same as the number of imported subjects of group 1.' ...
+		)
+
+assert(length(d2.get('TARGETS')) == gr2.get('SUB_DICT').get('LENGTH'), ...
+		[BRAPH2.STR ':NNDataPoint_CON_CLA:' BRAPH2.FAIL_TEST], ...
+		'NNDataPoint_CON_CLA does not construct the dataset correctly. The number of the targets should be the same as the number of imported subjects of group 2.' ...
+		)
+
+% Check whether the content of input for a single datapoint matches ¥\circled{10}\circlednote{10}{tests the value of each input from the data point matches the subject's connectivity data.}¥
+for index = 1:1:gr1.get('SUB_DICT').get('LENGTH')
+    individual_input = d1.get('DP_DICT').get('IT', index).get('INPUT');
+    known_input = {gr1.get('SUB_DICT').get('IT', index).get('CON')};
+
+    assert(isequal(individual_input, known_input), ...
+        [BRAPH2.STR ':NNDataPoint_CON_CLA:' BRAPH2.FAIL_TEST], ...
+        'NNDataPoint_CON_CLA does not construct the dataset correctly. The input value is not derived correctly.' ...
+        )
+end
+
+for index = 1:1:gr2.get('SUB_DICT').get('LENGTH')
+    individual_input = d2.get('DP_DICT').get('IT', index).get('INPUT');
+    known_input = {gr2.get('SUB_DICT').get('IT', index).get('CON')};
+
+    assert(isequal(individual_input, known_input), ...
+        [BRAPH2.STR ':NNDataPoint_CON_CLA:' BRAPH2.FAIL_TEST], ...
+        'NNDataPoint_CON_CLA does not construct the dataset correctly. The input value is not derived correctly.' ...
+        )
+end
+
+¤%%% ¡test! 
+%%%% ¡name!
+Example training-test ¤classification¤ ¥\circled{11}\circlednote{11}{executes the corresponding example scripts to ensure the functionalities.}¥
+%%%% ¡code!
+% ensure the example data is generated¤
+if ~isfile([fileparts(which('NNDataPoint_CON_CLA')) filesep 'Example data NN CLA CON XLS' filesep 'atlas.xlsx'])
+    test_NNDataPoint_CON_CLA % create example files
+¤end¤
+
+example_NN_CON_CLA
+
+\end{lstlisting}
+
+
+%%%%% %%%%% %%%%% %%%%% %%%%%
+\clearpage
+\section{Implementation of a Data Point with a Graph}
+\subsection{Graph Data Point for Regression (\code{NNDataPoint\_Graph\_REG})}
+
+Now we implement \code{NNDataPoint\_Graph\_REG} based on previous codes \code{NNDataPoint\_CON\_REG}.
+This neural network datapoint with graphs utilizes the adjacency matrix extracted from the derived graph of the subject. 
```

Comment:
**data point** (everywhere else you write data point with a space between data and point)

Reply:
<img width="355" alt="image" src="https://github.com/giovannivolpe/BRAPH-2-Matlab-beta/assets/30530538/adab5977-e997-471f-8d0d-89e02a732e4f">